share/normalize/oai.py

```diff
@@ -109,7 +109,15 @@ class OAIThroughTags(Parser):
 
 
 class OAICreativeWork(Parser):
-    schema = 'CreativeWork'
+    @property
+    def schema(self):
+        try:
+            resourceType = self.context['record']['metadata']['dc']['dc:type']
+            if resourceType == 'preprint':
+                return 'Preprint'
+            return 'CreativeWork'
+        except KeyError:
```

Comment:
You may want to catch `ValueError` here as well just in case one of the key returns `None`